contrib/exporters/secadvisor/mod/account_file.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package mod
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/spf13/viper"
+)
+
+func makeTimestamp() int64 {
+	return time.Now().UnixNano() / int64(time.Millisecond)
+}
+
+type accountFile struct {
+	filename string
+}
+
+type accountRecord struct {
+	StartupTime  int64 `json:"startup-time"`
+	StartupBytes int64 `json:"startup-bytes"`
+	UpdateTime   int64 `json:"update-time"`
+	UpdateBytes  int64 `json:"update-bytes"`
+}
+
+func (a *accountFile) read(filename string) *accountRecord {
+	record := &accountRecord{
+		StartupTime: makeTimestamp(),
+	}
+
+	file, err := os.Open(filename)
+	if err != nil {
+		return record
+	}
+	defer file.Close()
+
+	content, _ := ioutil.ReadAll(file)
```

Comment:
Need to add retry mechanism to read (and also to write) since we know that other processes are going to `sit` on this file in parallel to us 
(I also advice on this point to add dedicated test as this is a classic 'long term' failure of sure feature)

Reply:
I don't see a common case in which read may fail (as we write to a temporary location and then rename). Also I don't see a case in which readers can block the single writer. Are you able (via bash commands) to construct a case which requires addressing?